File: lib/libc/libc_hooks.c
#include "proto.h"


int _kill(int pid, int sig) {
    return 0;
}

int _write(int fd, const void *buf, size_t count)
{
    return 0;
}

void _exit(int status) {
    while (1) {
        ;
    }
}

int _getpid(void) {
    return 0;
}


void *_sbrk(int incr) {
  extern char _end;

  static unsigned char *heap = NULL;
  unsigned char *prev_heap;

  if (heap == NULL) heap = (unsigned char *) &_end;

  prev_heap = heap;
  heap += incr;

  return prev_heap;
}

File: Sources/lib/libc/libc_hooks.swift

@_cdecl("_kill")
public func _kill(_ pid: Int32, _ sig: Int32) -> Int32 {
    return 0
}

@_cdecl("_write")
public func _write(_ fd: Int32, _ buf: UnsafeRawPointer?, _ count: Int) -> Int32 {
    return 0
}

@_cdecl("_exit")
public func _exit(_ status: Int32) {
    while true {
        // Do nothing
    }
}

@_cdecl("_getpid")
public func _getpid() -> Int32 {
    return 0
}

@_cdecl("_sbrk")
public func _sbrk(_ incr: Int) -> UnsafeMutableRawPointer? {
    return nil
}
